plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java

```diff
@@ -0,0 +1,127 @@
+
+package com.google.cloud.tools.eclipse.appengine.newproject.maven;
+
+import org.apache.maven.archetype.catalog.Archetype;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.project.IProjectConfigurationManager;
+import org.eclipse.m2e.core.project.ProjectImportConfiguration;
+import org.eclipse.m2e.core.ui.internal.wizards.MappingDiscoveryJob;
+import org.eclipse.ui.actions.WorkspaceModifyOperation;
+
+import java.lang.reflect.InvocationTargetException;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CreateMavenBasedAppEngineStandardProject extends WorkspaceModifyOperation {
+  private static final Logger logger =
+      Logger.getLogger(CreateMavenBasedAppEngineStandardProject.class.getName());
+
+  IProjectConfigurationManager projectConfigurationManager =
+      MavenPlugin.getProjectConfigurationManager();
+
+  private String appEngineProjectId;
+  private String packageName;
+  private String artifactId;
+  private String groupId;
+  private IPath location;
+
+
+
+  @Override
+  protected void execute(IProgressMonitor monitor)
+      throws CoreException, InvocationTargetException, InterruptedException {
+    SubMonitor progress = SubMonitor.convert(monitor);
+    monitor.beginTask("Creating Maven AppEngine archetype", 100);
+
+    String appId = appEngineProjectId;
```

Comment:
It's an artefact of the archetype that we _have_ to provide a project id, but we're not requiring it of users.  Not entirely sure that the artifactId is the right choice to default to.